torch/csrc/jit/codegen/cuda/kernel_cache.cpp

```diff
@@ -250,15 +251,38 @@ std::vector<at::Tensor> FusionExecutorCache::runFusionWithInputs(
       // copy the fusion, since each FusionExecutor needs to manipulate the
       // fusion in order to generate kernel.
       Fusion fusion = *fusion_;
-      TensorView* red_tv = nullptr;
-      for (auto expr : fusion.exprs()) {
-        if (expr->getExprType().has_value() &&
-            expr->getExprType().value() == ExprType::ReductionOp) {
-          red_tv = expr->outputs()[0]->as<TensorView>();
-          break;
+
+      FusionGuard fg(&fusion);
+
+      TensorView* reduction_tv = nullptr;
+      // Use dependency check to find the reduction tv as it returns used values
+      // instead of exprs.
+      auto used_vals = DependencyCheck::getAllValsBetween(
+          {fusion.inputs().begin(), fusion.inputs().end()}, fusion.outputs());
```

Comment:
Does `getAllValsBetween(...)` have a smaller traverse space than `fusion.exprs()` because we might have trailing nodes after outputs? Just curious about the motivation of code change here.